app/src/main/java/com/example/whatsapp_clone/Model/User.java

```diff
@@ -14,12 +14,25 @@ public User(String username, String displayName, String profilePic) {
         this.profilePic = profilePic;
     }
 
+    public static class UserRegistration extends User {
+
+        String password;
+        public UserRegistration(String username,
+                                String password ,
+                                String displayName,
+                                String profilePic) {
+            super(username, displayName, profilePic);
+            this.password = password;
+        }
+    }
+
```

Comment:
if you're not going to teel use in the group chat what is the roll of this class please write us some doc for us to know what your planning

Reply:
Agree, 
This is the class the register fragment/ viewmodel need to create before performing the create user query